srml/contract/src/gas.rs

```diff
@@ -235,18 +228,16 @@ pub fn buy_gas<T: Trait>(
 }
 
 /// Refund the unused gas.
-pub fn refund_unused_gas<T: Trait>(transactor: &T::AccountId, gas_meter: GasMeter<T>) {
+pub fn refund_unused_gas<T: Trait>(transactor: &T::AccountId, gas_meter: GasMeter<T>) -> Result<(), &'static str> {
 	// Increase total spent gas.
 	// This cannot overflow, since `gas_spent` is never greater than `block_gas_limit`, which
 	// also has T::Gas type.
 	let gas_spent = <Module<T>>::gas_spent() + gas_meter.spent();
 	<GasSpent<T>>::put(gas_spent);
 
 	// Refund gas left by the price it was bought.
-	let b = <balances::Module<T>>::free_balance(transactor);
 	let refund = <T::Gas as As<T::Balance>>::as_(gas_meter.gas_left) * gas_meter.gas_price;
-	<balances::Module<T>>::set_free_balance(transactor, b + refund);
-	<balances::Module<T>>::increase_total_stake_by(refund);
+	<T as Trait>::ChargeFee::refund_fee(transactor, refund)
```

Comment:
If `refund_fee` always comes after `charge_fee` and the amount of refund the less that charge, then it can't fail. `refund_fee` has a `Result` return type here to make it more robust because we can't guarantee how it is called. Is it better to document this kind of expected way of usage, and get rid of `Result` return type?